File: Algorithms/Search algorithms/binary_search.cpp
int binarySearch(int arr[], int l, int r, int x) 
{ 
    if (r >= l) { 
        int mid = l + (r - l) / 2; 
  
        // If the element is present at the middle 

        if (arr[mid] == x) 
            return mid; 
  
        // If element is smaller than mid, then search in left subarray 
        if (arr[mid] > x) 
            return binarySearch(arr, l, mid - 1, x); 
  
        // Else search in right subarray 
        return binarySearch(arr, mid + 1, r, x); 
    } 
 
    return -1; 
} 

File: Algorithms/Search algorithms/binary_search.py
list1=["2","5","8","9"]
length=len(list1)

val=input()

def binarySearch(list1, l, r, x):

 if r >= l: 
 
  mid = l + (r - l) //2

  if list1[mid] == x:
   return mid
  
  elif list1[mid] > x: 
   return binarySearch(list1, l, mid - 1, x) 
  
  return binarySearch(list1, mid + 1, r, x) 
 
 return -1 
   
index=binarySearch(list1,0,length-1,val)   
      
if index ==-1:
 print("not present")
else:
 print("present at index:",index)
